packages/flutter/lib/src/scheduler/binding.dart

```diff
@@ -215,15 +215,53 @@ mixin SchedulerBinding on BindingBase {
 
   final List<TimingsCallback> _timingsCallbacks = <TimingsCallback>[];
 
-  /// Add a [TimingsCallback] that receives [FrameTiming] sent from the engine.
+  /// Add a [TimingsCallback] that receives [FrameTiming] sent from
+  /// the engine.
+  ///
+  /// This API enables applications to monitor their graphics
+  /// performance. Data from the engine is batched into lists of
+  /// [FrameTiming] objects which are reported approximately once a
+  /// second in release mode and approximately once every 100ms in
+  /// debug and profile builds. The list is sorted in ascending
+  /// chronological order (earliest frame first). The timing of the
+  /// first frame is sent immediately without batching.
+  ///
+  /// The data returned can be used to catch missed frames (by seeing
+  /// if [FrameTiming.buildDuration] or [FrameTiming.rasterDuration]
+  /// exceed the frame budget, e.g. 16ms at 60Hz), and to catch high
+  /// latency (by seeing if [FrameTiming.totalSpan] exceeds the frame
+  /// budget). It is possible for no frames to be missed but for the
+  /// latency to be more than one frame in the case where the Flutter
+  /// engine is pipelining the graphics updates, e.g. because the sum
+  /// of the [FrameTiming.buildDuration] and the
+  /// [FrameTiming.rasterDuration] together exceed the frame budget.
+  /// In those cases, animations will be smooth but touch input will
+  /// feel more sluggish.
+  /// 
+  /// Using [addTimingsCallback] is preferred over using
+  /// [Window.onReportTimings] directly because the
+  /// [Window.onReportTimings] API only allows one callback, which
+  /// prevents multiple libraries from registering listeners
+  /// simultaneously, while this API allows multiple callbacks to be
+  /// registered independently.
+  ///
+  /// This API is implemented in terms of [Window.onReportTimings].
+  /// In release builds, when no libraries have registered with this API, the
+  /// [Window.onReportTimings] callback is set to null, which reduces
```

Comment:
> which reduces the runtime overhead of performance tracking to approximately zero.

This sentence confused me. It seems to imply that you still get full performance tracking even when no callback is registered in release mode. Is that true? If so, why don't we use this penalty-free way of tracking in all run modes? Or should this make clearer, that if no callback is registered, performance tracking is more-or-less disabled and that's why its performance overhead is zero?

Reply:
tried to clarify